packages/react-scripts/scripts/test.js

```diff
@@ -20,21 +25,32 @@ require('dotenv').config({silent: true});
 
 const jest = require('jest');
 const argv = process.argv.slice(2);
+const debugFlag = require('../utils/getDebugFlag')(argv);
+const isDebug = !!debugFlag;
+const isRunInBand = argv.indexOf('--runInBand') > -1 || argv.indexOf('-i') > -1
 
 // Watch unless on CI or in coverage mode
-if (!process.env.CI && argv.indexOf('--coverage') < 0) {
+if (!process.env.CI && argv.indexOf('--coverage') < 0 && !isDebug) {
   argv.push('--watch');
 }
 
+// Force debug into single worker
+if (isDebug) {
+  if (!isRunInBand) {
+    argv.push('--runInBand')
+  }
+  argv.splice(argv.indexOf(debugFlag), 1)
```

Comment:
any of the node debug flags are not valid input to `jest`.  something counter convention i did in this PR was allow a debug flag (like `--debug-brk`) to enter via argv, where previously all of the other args were `jest` specific.  IMHO, this is OK.  the filtering may be cleaner towards the top-of-the-file